clients/api-d-roles/index.ts

```diff
@@ -0,0 +1,156 @@
+import { HttpServerError, HttpUnauthorizedError } from '#clients/exceptions';
+import routes from '#clients/routes';
+import constants from '#models/constants';
+import { Information } from '#models/exceptions';
+import httpClient, { httpGet, IDefaultRequestConfig } from '#utils/network';
+import { logInfoInSentry } from '#utils/sentry';
+import { URLSearchParams } from 'url';
+import {
+  IDRolesAuthTokenResponse,
+  IDRolesGroupSearchResponse,
+  IDRolesRoles,
+  IDRolesUser,
+} from './interface';
+
+/**
+ * D-Roles
+ * https://roles.preprod.data.gouv.fr/
+ */
+class DRolesAPIClient {
+  private _token: {
+    data: IDRolesAuthTokenResponse;
+    tokenExpiryTime: number;
+  } | null;
+
+  constructor(
+    private client_id: string | undefined,
+    private client_secret: string | undefined
+  ) {
+    if (
+      (!this.client_id || !this.client_secret) &&
+      process.env.NODE_ENV === 'production'
```

Comment:
Condition is a bit hard to read. I think the test on client_id and client_secret is enough

Reply:
I believe in staging we do have `process.env.NODE_ENV === 'production'`

I am not sure about this whole condition and how it will work in production